code/acting/src/acting/vehicle_controller.py

```diff
@@ -215,12 +223,24 @@ def __set_pure_pursuit_steer(self, data: Float32):
     def __set_stanley_steer(self, data: Float32):
         self.__stanley_steer = data.data
 
-    def __choose_controller(self) -> int:
+    def sigmoid(self, x: float):
+        temp_x = min(-25 * x, 25)
+        res = 1 / (1 + math.exp(temp_x))
+        return res
+
+    def __choose_controller(self) -> float:
         """
-        Chooses with steering controller to use
+        Returns the proportion of stanley to use.
+        Publishes the currently used controller
         :return:
         """
-        return STANLEY_CONTROLLER
+        res = self.sigmoid(self.__velocity - STANLEY_CONTROLLER_MIN_V)
+        if res >= 0.5:
```

Comment:
Maybe do the publishing where you do the logging in line 124 (or the other way around) to keep things together

Reply:
Good catch, fixed in [d556b8e](https://github.com/ll7/paf22/pull/210/commits/d556b8ea0a8a784c0dcf05d11e1a526768c9dfce)